citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -130,21 +131,29 @@
             logger.info("Not a valid URI: " + uri, e);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-
         entry = authorityService.create(entry, (IUser) authentication.getPrincipal());
         return new ResponseEntity<IAuthorityEntry>(entry, HttpStatus.OK);
     }
     
     @RequestMapping(value = "/auth/authority/add", method = RequestMethod.POST)
     public ResponseEntity<IAuthorityEntry> addAuthorityEntry(Authentication authentication,
-            @RequestParam("uri") String uri, @RequestParam("source") String source) {
+            @RequestParam("uri") String uri, @RequestParam("source") String source,
+            @RequestParam(defaultValue = "", required = false, value = "zoteroGroupId") String zoteroGroupId) {
         
         IAuthorityEntry entry = null;
         if(source != null && authoritySearchResult!= null && authoritySearchResult.containsKey(source)) {
             Optional<IAuthorityEntry> authorityEntry = authoritySearchResult.get(source).getFoundAuthorities().stream()
                     .filter(authority -> authority.getUri().equals(uri)).findFirst();
             
             if (authorityEntry.isPresent()) {
+                if(!zoteroGroupId.isEmpty()) {
+                    List<Long> authorityGroups = authorityEntry.get().getGroups();
+                    if(authorityGroups == null) {
+                        authorityGroups = new ArrayList<>();
+                    }
+                    authorityGroups.add(Long.valueOf(zoteroGroupId));
+                    authorityEntry.get().setGroups(authorityGroups);
```

Comment:
if `authorityGroups` is already initialized this line doesn't really do anything. In your if statement I would check `authorityEntry.get().getGroups() == null` and if so, initialize the list. Then just add the new group id. Also, make sure there are no duplicates in the list.